components/async-status-mgt/org.wso2.carbon.identity.framework.async.status.mgt/src/main/java/org/wso2/carbon/identity/framework/async/status/mgt/dao/AsyncStatusMgtDAOImpl.java

```diff
@@ -0,0 +1,880 @@
+                    responseContext.setResidentOrgId(
+                            resultSet.getString(
+                                    SQLConstants.OperationStatusTableColumns.IDN_OPERATION_INITIATED_ORG_ID));
+                    responseContext.setInitiatorId(
+                            resultSet.getString(
+                                    SQLConstants.OperationStatusTableColumns.IDN_OPERATION_INITIATED_USER_ID));
+                    responseContext.setOperationStatus(
+                            resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_STATUS));
+                    responseContext.setOperationPolicy(
+                            resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_OPERATION_POLICY));
+                    responseContext.setCreatedTime(
+                            Timestamp.valueOf(
+                                    resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_CREATED_AT)));
+                    responseContext.setModifiedTime(
+                            Timestamp.valueOf(
+                                    resultSet.getString(SQLConstants.OperationStatusTableColumns.IDN_LAST_MODIFIED)));
+                    responseContexts.add(responseContext);
+                }
+            }
+        } catch (SQLException e) {
+            String errorMessage =
+                    "Error fetching async operation status for subject: " + operationSubjectId + " within " + days +
+                            " days.";
+            LOGGER.info(errorMessage + e);
+            throw new RuntimeException(errorMessage, e);
+        }
+        LOGGER.info(
+                "Fetching async operation status for subject: " + operationSubjectId
+                        + " within " + days + " days.");
+        return responseContexts;
+    }
+
+    @Override
+    public void updateAsyncOperationStatus(String operationId, String status) {
+
+        String currentTimestamp = new Timestamp(new Date().getTime()).toString();
+
+        String sql = "UPDATE IDN_ASYNC_OPERATION_STATUS " +
+                "SET IDN_OPERATION_STATUS = ?, " +
+                "IDN_LAST_MODIFIED = ? " +
+                "WHERE IDN_OPERATION_ID = ?";
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setString(1, status);
+            statement.setString(2, currentTimestamp);
+            statement.setString(3, operationId);
+
+            statement.executeUpdate();
+
+        } catch (SQLException e) {
+            throw new RuntimeException(e);  //TODO: exception
+        }
+    }
+
+    private boolean hasExistingRecords(Connection connection, String operationType, String operationSubjectId)
+            throws SQLException {
+
+        String checkSql = "SELECT COUNT(*) FROM IDN_ASYNC_OPERATION_STATUS " +
```

Comment:
```suggestion
        String checkSql = "SELECT 1  FROM IDN_ASYNC_OPERATION_STATUS " +
```

If we are checking only the existence, this can be more efficient rather than using count